tools/ib-test/ib_test.cpp

```diff
@@ -333,16 +353,30 @@ int main(int argc,char* argv[])
       netConnectAddr.sin_port = htons(23456);
 
     netConnectAddr.sin_family = AF_INET;
-    printf("Connecting to %s:%s\n", ip, port);
+    printf("Connecting to %s:%d\n", ip, ntohs(netConnectAddr.sin_port));
 
     printf("GDR Read %s\n", use_gdr_read ? "enabled" : "disabled");
 
-    NCCLCHECK(ncclCudaCalloc(&sendDevBuffer, sendBuffSize, 1));
-    NCCLCHECK(ncclCudaHostCalloc(&sendHostBuffer, sendBuffSize));
-    int status[1] = {-1};
-    if (!move_pages(0, 1, (void **)&sendHostBuffer, NULL, status, 0))
-      printf("Allocated sendHostBuffer %p of %d bytes on node %d, sliceSteps %d\n",
-        sendHostBuffer, sendBuffSize, status[0], sliceSteps);
+    if (use_gdr_read) {
+      NCCLCHECK(ncclCudaCalloc(&sendDevBuffer, sendBuffSize, 1));
+      printf("Allocated sendDevBuffer %p of %d bytes, sliceSteps %d\n",
+                sendDevBuffer, sendBuffSize, sliceSteps);
+    }
+    else {
+#ifdef USE_MEMALIGN
+      int page_size = getpagesize();
+      posix_memalign((void **)&sendHostBuffer, page_size, sendBuffSize);
+      hipHostRegister(sendHostBuffer, sendBuffSize, hipHostRegisterMapped);
```

Comment:
Is there any particular reason why we don't have CUDACHECK() around any the hipHostRegister and hipHostGetDevicePointer calls?

Reply:
good catch! fixed.